### ISSUE TITLE
Compilation for SPI fails

### ISSUE THREAD

**Issue by kolins-cz** [latonita/esphome-ina228#2]
Hello,
Not sure what happens, but code for SPI version fails to compile. Code (shortened):

```
esphome:
  name: solar-battery
  friendly_name: Solar Battery
  comment: ${device_description}
esp32:
  board: esp32dev
  framework:
    type: esp-idf

external_components:
  - source: github://syssi/esphome-jk-bms@main
  - source: github://syssi/esphome-jbd-bms@main
  - source: github://latonita/esphome-ina228@main
    components: [ina2xx_base, ina2xx_i2c]
    refresh: 0s

spi:
    clk_pin: GPIO27
    mosi_pin: GPIO26
    miso_pin: GPIO25
i2c:
  sda: GPIO18
  scl: GPIO19

sensor:
  - platform: ina2xx_spi
    cs_pin: GPIO33
    shunt_resistance: 0.001130 ohm
    max_current: 40 A
    adc_range: 0
    adc_time: 4120 us
    adc_averaging: 128
    temperature_coefficient: 50
```

![image](https://github.com/latonita/esphome-ina228/assets/8409715/76f52df0-a803-4361-af09-9a1f42260685)

If I change sensor platform to `ina2xx_i2c` and add replace `cs_pin: GPIO33` with `address: 0x40` code compiles

Also, there seems to be missing support for multiple SPI and I2C buses.

**Comment by kolins-cz** [comment]
Found the bug. Your example for SPI is incorrect. This line fixes it:
`components: [ina2xx_base, ina2xx_spi]`

**Comment by latonita** [comment]
Thanks for finding this, yes, definitely there shall be SPI not I2C. I will fix the example.

Please report if there are any other issues. SPI version was not tested, all the people had I2C versions of the sensor.

**Comment by kolins-cz** [comment]
Installed INA229 in application and it does work at first try. I would not recommend SPI version as there are two more wires. I got SPI version during chip crisis, when INA228 was unobtainium.

I yet have to understand (and read datasheet again) what Energy and Charge registers does. 
Right now Energy goes up and down same as Charge (I am only discharging right now). 

![image](https://github.com/latonita/esphome-ina228/assets/8409715/ee81480d-3304-4e99-aff7-922978276829)
![image](https://github.com/latonita/esphome-ina228/assets/8409715/304ee8ce-66b5-4143-b927-037a1fc99e18)

Also it would be nice to limit default number of decimals to some sane value. I did it in YAML by `accuracy_decimals: 2` for each sensor.
